docs/specifications/tests/DNSQueryAndResponseDefaults.md

```diff
@@ -0,0 +1,165 @@
+
+
+## Default handling of a *DNS Response*
+
+A *DNS Response* is a response to a *DNS Query*. Unless specified in the test
+case specification, the items in the response are handled as listed. If a
+*Response Item* is specified as "fixed" (with an "X" in that column) then the
+requirement, as specified under "Default handling", must be met for the response
+to be considered to be a DNS response.
+
+|Response Item |Default handling                          | Fixed | Comment              |
+|:-------------|:-----------------------------------------|:------|:---------------------|
+|OpCode        | Require value to be "response"           | X     |                      |
+|QR flag       | Require flag to be set                   | X     |                      |
+|AA flag       | -                                        |       | Defined in test case |
+|TC flag       | Re-query over TCP if set                 |       |                      |
+|RD flag       | ignore                                   |       |                      |
+|RA flag       | ignore                                   |       |                      |
+|AD flag       | ignore                                   |       |                      |
+|CD flag       | ignore                                   |       |                      |
+|RCODE         | -                                        |       | Defined in test case |
+|Query name    | ignore                                   |       |                      |
+|Query type    | ignore                                   |       |                      |
+|Query class   | Require value to be same as in the query |       | Normally "IN"        |
+|EDNS          | ignore                                   |       |                      |
+
+* Check against query name and query type is, by default, done against the values
+  in the question section in the query, not in the response.
+  
+* When fetching records from the answer section, these are the default criteria:
+  * Only records matching *Query Name* and *Query Type* are fetched. Any
+    RRSIG records meeting the next criterion are also fetched.
+  * RRSIG records matching *Query Name* and covering *Query Type* are
+    fetched.
+  * CNAME records are ignored unless *Query Type* is CNAME.
+
+* When the test case specification states that a CNAME chain is to be followed,
+  the default handling is to look for a valid CNAME chain in the answer section
+  and fetch all records in that chain. The records in the chain are arranged in
+  the logical order. These are the default criteria for a CNAME chain to be
+  valid:
+  * The first CNAME in the chain must match *Query Name*.
+  * The last record in the chain must either be a CNAME or a record matching
+    *Query Type*.
+  * For each owner name of the CNAME records in the chain there must not be any
+    additional CNAME records in the answer section (only one CNAME record per
+    owner name).
```

Comment:
I'm curious about an edge case here. Let's say we're following a CNAME chain and the query name is example.com. The answer section contains the following CNAME records:
```
error.com    3600  IN  CNAME  example.com
example.com  3600  IN  CNAME  other.com
```
The way I read the specification there's one chain in the answer section, and the first CNAME is error.com, which is different from the query name, so none of these records should be fetched. Is that the intended behavior?

I believe the "The first CNAME in the chain must match Query Name." item should be removed from the list of criteria and added as a definition instead.

Reply:
> I'm curious about an edge case here. Let's say we're following a CNAME chain and the query name is example.com. The answer section contains the following CNAME records:
> 
> ```
> error.com    3600  IN  CNAME  example.com
> example.com  3600  IN  CNAME  other.com
> ```
> 
> The way I read the specification there's one chain in the answer section, and the first CNAME is error.com, which is different from the query name, so none of these records should be fetched. Is that the intended behavior?

Yes.

> I believe the "The first CNAME in the chain must match Query Name." item should be removed from the list of criteria and added as a definition instead.

I disagree, but I will change the wording somewhat to make intended behavior clear.